### ISSUE TITLE
Can't add as extra-dep in stack.yaml

### ISSUE THREAD

**Issue by o1lo01ol1o** [AccelerateHS/accelerate-llvm#6]
I'm trying to link the current accelerate-llvm master in a project but stack keeps complaining that it can't find a .cabal file in the downloaded directory.  Is this something that can be updated?  If not, is there a workaround?

**Comment by tmcdonell** [comment]
I guess you left out the `subdirs` section? e.g.:

```
- location:
    git:    https://github.com/AccelerateHS/accelerate-llvm.git
    commit: 987bff46f146b718a1da60c194eceff4b91a843f
  extra-dep: true
  subdirs:
    - accelerate-llvm
    - accelerate-llvm-native
    - accelerate-llvm-ptx
```

**Comment by o1lo01ol1o** [comment]
Yes, you were right, my mistake.  

However, now I seem to have a new error that may be caused by my llvm version (which my project requires to be >3.5). Is this a version error?

```
--  While building package accelerate-llvm-2.0.0.0 using:
      /Users/timpierson/.stack/setup-exe-cache/x86_64-osx/setup-Simple-Cabal-1.22.5.0-ghc-7.10.3 --builddir=.stack-work/dist/x86_64-osx/Cabal-1.22.5.0 build lib:accelerate-llvm --ghc-options " -ddump-hi -ddump-to-file"
    Process exited with code: ExitFailure 1
    Logs have been written to: /Users/timpierson/subhask/.stack-work/logs/accelerate-llvm-2.0.0.0.log

    Preprocessing library accelerate-llvm-2.0.0.0...
    [ 1 of 45] Compiling Data.Range.Range ( Data/Range/Range.hs, .stack-work/dist/x86_64-osx/Cabal-1.22.5.0/build/Data/Range/Range.o )
    ghc: 
    lookupSymbol failed in relocateSection (RELOC_GOT)
    /usr/local/lib/libLLVMSupport.a: unknown symbol `___dso_handle'
    ghc: unable to load package `llvm-general-3.5.1.2'
```

**Comment by tmcdonell** [comment]
You probably didn't tell `llvm-general` to use the shared library (or your LLVM install wasn't configured to produce it). Try adding this to your `stack.yaml`:

```
flags:
  llvm-general:
    shared-llvm: true
```

`llvm-general` is necessarily strict on what versions of llvm it compiles with, so if your project requires llvm > 3.5 you may run into some problems, as the current released version uses llvm-3.5.

**Comment by o1lo01ol1o** [comment]
You were correct again, however, it wasn't because the listed flags, it was because configuring LLVM with cmake does not actually flag the correct shared files for compilation that llvm-general requires.
for posterity, use the following:

```
    $ wget http://llvm.org/releases/3.5.2/llvm-[xxx].src.tar.xz
    $ tar -xf llvm-[xxx].src.tar.xz
    $ cd llvm-[xxx]
    $ mkdir build
    $ cd build
    $ ../configure --enable-shared --enable-optimized
    $ make -j5
    $ sudo make install
    ```

I'm running into another accelerate issue, but I'll make an issue in the main repo.  Thanks for your help.
